engine/StoryState.js

```diff
@@ -385,15 +386,20 @@ export class StoryState{
 			var listText = this.TrySplittingHeadTailWhitespace(text);
 			if (listText != null) {
 				listText.forEach(textObj => {
-					this.PushToOutputStreamIndividual(textObj);	
+					this.PushToOutputStreamIndividual(textObj);
 				});
+				this.OutputStreamDirty();
 				return;
 			}
 		}
 
 		this.PushToOutputStreamIndividual(obj);
 		this.OutputStreamDirty();
 	}
+	PopFromOutputStream(count){
+		this.outputStream.splice(this.outputStream.length - count, count);
```

Comment:
Same here with `splice`, it should do exactly the same as `RemoveRange` but a double check would not be bad.